internal/provider/datasources/service_account_test.go

```diff
@@ -0,0 +1,45 @@
+package datasources_test
```

Comment:
@armalite i added an acceptance test here against a dedicated testing environment, which verified your creation / deletion logic for the `prefect_service_account` resource as well as catching a few small bugs with the datasource 🙌